rest-server/src/models/job.js

```diff
@@ -94,11 +95,41 @@ class Job {
             next(job, new Error('job not found'));
           } else {
             if (framework.frameworkStatus) {
+              let jobState = "";
+              const frameworkState = framework.frameworkStatus.frameworkState;
+              if (
+                  frameworkState == "FRAMEWORK_WAITING" ||
+                  frameworkState == "APPLICATION_CREATED" ||
+                  frameworkState == "APPLICATION_LAUNCHED" ||
+                  frameworkState == "APPLICATION_WAITING"
+                ) {
+                jobState = "WAITING";
+              } else if (
+                  frameworkState == "APPLICATION_RUNNING" ||
+                  frameworkState == "APPLICATION_RETRIEVING_DIAGNOSTICS" ||
+                  frameworkState == "APPLICATION_COMPLETED"
+                ) {
+                jobState = "RUNNING";
+              } else if (
+                  frameworkState == "FRAMEWORK_COMPLETED"
+                ) {
+                jobState = "COMPLETED";
+              }
+              let jobRetryCount = 0;
+              const jobRetryCountInfo = framework.frameworkStatus.frameworkRetryPolicyState;
+              if (jobRetryCountInfo) {
+                jobRetryCount =
+                  jobRetryCountInfo.transientNormalRetriedCount +
+                  jobRetryCountInfo.transientConflictRetriedCount +
+                  jobRetryCountInfo.nonTransientRetriedCount +
+                  jobRetryCountInfo.unKnownRetriedCount;
```

Comment:
check before add

Reply:
@abuccts : I asked Yuqi. These fields will always be there. Then do we still need to check them?